keps/sig-storage/2639-secret-protection/README.md

```diff
@@ -0,0 +1,450 @@
+    - If no `secret-protection-volumesnapshot-controller` exists but `snapshot.storage.kubernetes.io/secret-protection` finalizer is added to the secrets, no one remove the finalizer. Therefore, user needs to remove the `kubernetes.io/secret-protection` finalizer from all the secrets manually.
+
+### Version Skew Strategy
+
+- As for the difference between in-tree components and out-of-tree components, they work independently, so they won't affect each other,
+- As for in-tree components, the protection logic involves only in-tree admission controller and in-tree secret-protection-controller, so version skew won't happen unless these components are available with different versions,
+- As for out-of-tree components, the protection logic involves only out-of-tree admission controller and out-of-tree secret-protection-volumesnapshot-controller, so version skew won't happen unless these components are available with different versions,
+- As for resources, CSI Volume and CSI Snapshot are involved, changes in the API/CRD of these resources especially for their secret fields might cause issue. Howerver, this should be compatibility issue for these API/CRDs.
+
+## Production Readiness Review Questionnaire
+
+<!--
+
+Production readiness reviews are intended to ensure that features merging into
+Kubernetes are observable, scalable and supportable; can be safely operated in
+production environments, and can be disabled or rolled back in the event they
+cause increased failures in production. See more in the PRR KEP at
+https://git.k8s.io/enhancements/keps/sig-architecture/1194-prod-readiness.
+
+The production readiness review questionnaire must be completed and approved
+for the KEP to move to `implementable` status and be included in the release.
+
+In some cases, the questions below should also have answers in `kep.yaml`. This
+is to enable automation to verify the presence of the review, and to reduce review
+burden and latency.
+
+The KEP must have a approver from the
+[`prod-readiness-approvers`](http://git.k8s.io/enhancements/OWNERS_ALIASES)
+team. Please reach out on the
+[#prod-readiness](https://kubernetes.slack.com/archives/CPNHUMN74) channel if
+you need any help or guidance.
+-->
+
+### Feature Enablement and Rollback
+###### How can this feature be enabled / disabled in a live cluster?
+
+- [x] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name: SecretInUseProtection
+  - Components depending on the feature gate:
+    - kube-controller-manager (in-tree)
+    - secret-protection-controllerr (in-tree)
+    - storageobjectinuseprotection admission plugin (in-tree)
+    - secret-protection-volumesnapshot-controller (out-of-tree)
+    - storageobjectinuseprotection volumesnapshot admission plugin (out-of-tree)
```

Comment:
I'm not sure how you can use the in-tree feature gate on kube-controller-manager to control these out of tree components?  How do you check it?  I think for out-of-tree components, we can just say that it is enabled when it is deployed.

storageobjectinuseprotection -> secretinuseprotection

Reply:
I thought that [this](https://github.com/kubernetes-csi/external-provisioner/blob/master/cmd/csi-provisioner/csi-provisioner.go#L125) is sharing the feature gate, but after looking more in details, it seems that it isn't actually sharing the same values, instead it is just passing the another values with the same key. "just say that it is enabled when it is deployed" looks better to avoid confusion.